src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -29,27 +29,24 @@ public static class TarUtils
             string inputPath,
             CancellationToken cancellationToken)
         {
-            var archiveFile = Path.Combine(Path.GetTempPath(), archiveFileName);
-            if (File.Exists(archiveFile))
+            if(File.Exists(inputPath))
             {
-                File.Delete(archiveFile);
+                throw new InvalidOperationException($"Please specify path to a directory. {inputPath} is a file.");
```

Comment:
>{inputPath} is a file [](start = 90, length = 21)

Also specify that file path is not allowed.

Reply:
Done.